app/clients/cornershop_client.rb

```diff
@@ -0,0 +1,76 @@
+class CornershopClient
+  BASE_URL = 'https://cornershopapp.com/api/v2/branches/search'
+  DEFAULT_COUNTRY_CODE = 'CL'
+
+  def products_by_query(query: nil, locality: 'Providencia')
+    response = HTTParty.get(
+      products_url(query, locality)
+    )
+
+    products_found_to_json(response['results'])
+  end
+
+  private
+
+  def products_found_to_json(stores)
+    products = []
+    stores.each do |store_result|
+      store = store_result['store']
+      store_search_result = store_result['search_result']['aisles'].first
+
+      provider_name = get_store_name(store)
+      provider = Provider.find_or_create_by(name: provider_name)
+
+      products << {
+        provider: provider.as_json.with_indifferent_access,
+        products: products_found_by_store(store_search_result)
+      }
+    end
+
+    products.flatten
+  end
+
+  def products_found_by_store(store_search_result)
+    store_search_result['products'].map do |product|
+      next unless product['purchasable']
+
+      {
+        price: get_price(product),
+        measure: get_measure(product),
+        package: get_package(product),
+        name: get_name(product),
+        img_url: get_image_url(product),
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.